packages/aws-rfdk/lib/core/lib/mountable-filesystem.ts

```diff
@@ -14,7 +17,7 @@ import { IScriptHost } from './script-assets';
  * {@link https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.Instance.html|EC2 Instance}
  * or an {@link https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-autoscaling.AutoScalingGroup.html|EC2 Auto Scaling Group}
  */
-export interface IMountingInstance extends IConnectable, IScriptHost {
+export interface IMountingInstance extends IConnectable, IConstruct, IScriptHost {
```

Comment:
I believe it's for the [node dependency addition](https://github.com/aws/aws-rfdk/pull/339/files#diff-4add095eaf52008f1b5fb6476d138899d4583b2dc36b57bddd9e223a1ea22fd8R95)
```
target.node.addDependency(this.props.filesystem.mountTargetsAvailable);
```

Reply:
I snuck in a change to ensure that EFS mount targets are available before mounting the EFS. See `packages/aws-rfdk/lib/core/lib/mountable-efs.ts`:

```ts
target.node.addDependency(this.props.filesystem.mountTargetsAvailable);
```

According to [the `FileSystem.mountTargetsAvailable` API docs](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-efs.FileSystem.html#mounttargetsavailablespan-classapi-icon-api-icon-experimental-titlethis-api-element-is-experimental-it-may-change-without-noticespan):

>   Dependable that can be depended upon to ensure the mount targets of the filesystem are ready.

Seems like we should have this for cases when the EFS is in the same stack as the host where it is mounted. Without it, the CloudFormation deployment might have a race condition between the mount targets and the host instance since the host's user data only references the file-system itself. I don't think this further constrains IMountingInstance, since we cannot add user data to a non-construct resource.

Reply:
Gotcha. Could the dependency code be conditional? If the thing isn't a construct, then there's no need for a dep.
e.g. Something like
```
target.node?.addDependency(....)
```